test/deploy-time-budget.sh

```
@@ -0,0 +1,148 @@
+
+MAX_DEPLOY_SECONDS="${MAX_DEPLOY_SECONDS:-45}"
+exit_code=0
+
+# Find a MySQL version >= 8.4 to test with (these are the versions
+# most affected by the #131 race condition).
+function find_test_version {
+    if [ ! -d "$SANDBOX_BINARY" ]; then
+        echo "# SANDBOX_BINARY ($SANDBOX_BINARY) not found"
+        return 1
+    fi
+    versions=$(ls "$SANDBOX_BINARY" | grep -E '^[0-9]+\.[0-9]+\.[0-9]+$' | sort -t. -k1,1n -k2,2n -k3,3n)
```

Shellcheck (0.11.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)